### DISCUSSION TITLE
Getting versioned sidebars to work

### DISCUSSION THREAD

**Discussion by koebi** [IMB11/vitepress-versioning-plugin#12] [Q&A] [answered]
Hey,

I'm having a bit of trouble getting versioned sidebars to work.
Everything else seems to work - version switching works and links on pages link to the correct version.

However, the sidebar seems to be the one from the latest version.

I'm guessing that something with the sidebar not being an array, or an object, or the versioned sidebar being wrong, is the culprit here, but I can't figure it out…

The project progress can be found here:
https://github.com/GIScience/openrouteservice/pull/1919

Grateful for any pointers :)

Best regards,
Jakob

**Comment by TheGreatRefrigerator** [comment, marked as answer]
After some debugging, it seems, the generateVersionSidebars function expects locales to be defined.
Otherwise it won't do anything.
Didn't work to add a locales object in `.vitepress/config.js` and moving the `themeConfig`, `title` and `description` there.

**Comment by koebi** [comment]
This sounds like a bug to me… @IMB11 should I open an issue? Or is this discussion enough?

**Comment by koebi** [comment]
I think I have an inkling on what is going on here:

As by the plugin docs, the files of the versions live in `versions/${version}/`.
The sidebar section generated for a version is `${version}/`.
The plugin generates a few rewrites from `versions/${version}/…` to `$version/…`.

Now the sidebar isn't displayed, because as per the [vitepress docs](https://vitepress.dev/reference/default-theme-sidebar#multiple-sidebars):
> // This sidebar gets displayed when a user // is on `config` directory.

i.e. the sidebar section and the *directory* should match up - which they don't, because of the rewrites.

Removing the rewrites and changing the sidebar section to `/versions/${version}/` displays the correct sidebar - but makes the version switcher not work anymore…

I'm unsure now as to whether this is a bug in vitepress…

**Comment by koebi** [comment]
…it seems this is a simple error in the generated sidebar key, see https://github.com/IMB11/vitepress-versioning-plugin/pull/13

**Comment by IMB11** [comment]
I'm really sorry if I haven't been contributing/maintaining this project very well, I don't want to take the pressure away from the vitepress developer to produce an actual solution to versioning rather than off-loading the problem to 3rd party libraries.
